raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -2255,9 +2255,9 @@ public void resign(int epoch) {
 
     @Override
     public Optional<SnapshotWriter<T>> createSnapshot(long committedOffset, int committedEpoch) {
-        return log.createNewSnapshot(
-            new OffsetAndEpoch(committedOffset + 1, committedEpoch)
-        ).map(snapshot -> {
+        final OffsetAndEpoch snapshotId = new OffsetAndEpoch(committedOffset + 1, committedEpoch);
+        validateSnapshotId(snapshotId);
```

Comment:
@feyman2016 Take a look at the implementation for `KafkaMetadataLog::createNewSnapshot` in trunk. I think that function now implement this check.

The check that it doesn't do explicitly is https://github.com/apache/kafka/pull/10593/files#diff-1da15c51e641ea46ea5c86201ab8f21cfee9e7c575102a39c7bae0d5ffd7de39R2281-R2285

I think it is okay to skip this check because the quorum epoch is always greater than or equal to the largest epoch in the log. I think we already have this check in `createNewSnapshot` that compares the snapshot id against the log: https://github.com/apache/kafka/pull/10593/files#diff-1da15c51e641ea46ea5c86201ab8f21cfee9e7c575102a39c7bae0d5ffd7de39R2286-R2290

Please confirm but I think we can remove this check since we have something similar in `createNewSnapshot` in trunk. I should keep your improvement to the tests.

Apologies once again for coding and merging this functionality but I needed this changes as part of https://github.com/apache/kafka/pull/10786

Reply:
@jsancio Totally understood, I just confirmed the validation logic has been covered already as you mentioned, so just kept the test.